src/tools/rbd_mirror/image_replayer/PrepareRemoteImageRequest.cc

```diff
@@ -129,20 +129,19 @@ void PrepareRemoteImageRequest<I>::handle_get_client(int r) {
 
   if (r == -ENOENT) {
     dout(10) << "client not registered" << dendl;
+    register_client();
   } else if (r < 0) {
     derr << "failed to retrieve client: " << cpp_strerror(r) << dendl;
     finish(r);
+  } else if (!decode_client_meta()) {
+    // require operator intervention since the data is corrupt
+    finish(-EBADMSG);
     return;
```

Comment:
nit: `return` is not needed (just to be consistent with other clauses, or add it to every clause).